backend/src/modules/event/repositories/event.repository.ts

```diff
@@ -1,24 +1,72 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
+import { OrderDirection } from 'src/common/enums/order-direction.enum';
+import {
+  Pagination,
+  RepositoryWithPagination,
+} from 'src/infrastructure/base/repository-with-pagination.class';
 import { ActivityTypeTransformer } from 'src/modules/activity-type/models/activity-type.model';
 import { OrganizationStructureTransformer } from 'src/modules/organization/models/organization-structure.model';
 import { Repository } from 'typeorm';
 import { EventEntity } from '../entities/event.entity';
+import { EventTime } from '../enums/event-time.enum';
 import { IEventRepository } from '../interfaces/event-repository.interface';
 import {
   CreateEventOptions,
   EventModelTransformer,
+  FindManyEventOptions,
   IEventModel,
   UpdateEventOptions,
   UpdateStatusOptions,
 } from '../models/event.model';
 
 @Injectable()
-export class EventRepository implements IEventRepository {
+export class EventRepository
+  extends RepositoryWithPagination<EventEntity>
+  implements IEventRepository
+{
   constructor(
     @InjectRepository(EventEntity)
     private readonly eventRepository: Repository<EventEntity>,
-  ) {}
+  ) {
+    super(eventRepository);
+  }
+
+  async getMany(
+    findOptions: FindManyEventOptions,
+  ): Promise<Pagination<IEventModel>> {
+    const { eventTime, organizationId, orderBy, orderDirection } = findOptions;
+
+    const query = this.eventRepository
+      .createQueryBuilder('event')
+      .leftJoinAndMapMany('event.targets', 'event.targets', 'targets')
+      .leftJoinAndMapMany('event.tasks', 'event.tasks', 'tasks')
+      .leftJoinAndMapMany('event.eventRSVPs', 'event.eventRSVPs', 'eventRSVPs')
+      .select()
+      .where('event.organizationId = :organizationId', { organizationId })
+      .orderBy(
+        this.buildOrderByQuery(orderBy || 'createdOn', 'event'),
+        orderDirection || OrderDirection.ASC,
+      );
+
+    const currentDate = new Date();
+    if (eventTime === EventTime.OPEN) {
+      query.andWhere('event.endDate > :currentDate', {
```

Comment:
Ai testat timezone? Sa nu avem probleme cu evenimente create la 00:00 care au localtime 10:00PM. Sau macar sa-i spui lui Mario sa aiba asta in vedere.

Altfel puteam sa folosim `NOW()` din Postgres, dar nu stiu ce TIMEZONE are ala.